ndb_api/db_request_types.go

```diff
@@ -29,12 +29,13 @@ type DatabaseProvisionRequest struct {
 	CreateDbServer           bool             `json:"createDbserver"`
 	NodeCount                int              `json:"nodeCount"`
 	NxClusterId              string           `json:"nxClusterId"`
-	SSHPublicKey             string           `json:"sshPublicKey"`
+	SSHPublicKey             string           `json:"sshPublicKey,omitempty"`
 	Clustered                bool             `json:"clustered"`
 	AutoTuneStagingDrive     bool             `json:"autoTuneStagingDrive"`
 	TimeMachineInfo          TimeMachineInfo  `json:"timeMachineInfo"`
 	ActionArguments          []ActionArgument `json:"actionArguments"`
 	Nodes                    []Node           `json:"nodes"`
+	DatabaseName             string           `json:"databaseName,omitempty"`
```

Comment:
same here

Reply:
This field is required for MSSQL Databases only. We could discuss offline how we can validate these engine specific fields.